src/store/modules/request.js

```diff
@@ -308,9 +308,10 @@ export default {
         const { cancelId } = requestsForKey;
 
         const baseConfig = { method: 'GET', url };
-        baseConfig.headers = extended
-          ? { ...headers, 'X-Extended-Metadata': 'true' }
-          : headers;
+        if (extended)
+          baseConfig.headers = { ...headers, 'X-Extended-Metadata': 'true' };
+        else if (headers != null)
+          baseConfig.headers = headers;
```

Comment:
This makes it so that `baseConfig.headers` is not set at all if the `headers` option was not specified. I noticed that it was being set while working on the new tests, because (I believe) the only place it comes up right now is the request to restore the session.